sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -713,7 +713,10 @@ fixed topology that can be submitted for execution on any queue sharing the
 context associated with the graph. It is valid to call this method multiple times
 to create subsequent executable graphs. It is also valid to continue to add new
 nodes to the modifiable graph instance after calling this function. It is valid
-to finalize an empty graph instance with no recorded commands.
+to finalize an empty graph instance with no recorded commands. The created
+executable graph(s) should not be executed simultaneously if at least one
+execution contains a write operation to a memory address accessed by another
+execution, it is undefined behavior to do so.
```

Comment:
This language seems a bit low level to me given that the user is not in direct control of the execution of a graph - that is ultimately decided by the SYCL runtime scheduler and backend. In the case of graphs which use buffers with write access, the scheduler will create dependencies between graph submissions anyway and prevent simultaneous execution to correctly enforce dependencies. For USM this seems to be no different than normal usage - the user is required to enforce dependencies themselves with events to ensure correct behaviour. 

It seems inappropriate to me to categorize incorrect usage of USM as undefined behaviour for graphs specifically, unless I am misunderstanding something here.

Reply:
agree that the buffer/access case and USM case do not have meaningful difference here.  For the USM case, the dependency is achieved by in-order queue internally or by user explicitly.

The change here is about the concurrent execution, suppose we have a recorded graph with 3 kernels A, B and C, and kernel A reads from buffer (or USM) a and writes to b,  kernel B reads from b and writes to c, kernel C reads from c and writes to d. And the dependency is correctly set between kernels, and a, b, c and d could be USMs or buffers.

Now we have two threads to execute the graph simultaneously. For a simplified timing below:
step1):  in thread 1,  A reads from a and writes to b
step2):  in thread 2,  a is filled with new data
step3):  in thread 2,  A reads from a and writes to b
step4):  in thread 1,  B reads from b with wrong data (overwritten by thread 2)

question: will the SYCL runtime scheduler block thread 2 at step3?

this PR is to let the user know the above behavior is undefined.

btw, we'd remove 'address' in the description to also cover the buffer case.

Reply:
> another question is: how could the two CPU threads prepare the content of 'a' before the execution, if the concurrent execution is supported?

for this question, and also the fact that there should be an input and/or an output for the graph execution, I'll refine the wording to remove the reason since it is not easy to explain for all the cases.

@Bensuo any comment? thanks.